docs/user-guide/federation/federated-services.md

```diff
@@ -0,0 +1,234 @@
+3. Similarly, if no healthy shards exist in the enclosing region, the search progresses further afield. In the europe-west1-d availability zone, there are no healthy backends, so queries are redirected to the broader europe-west1 region (which also has no healthy backends), and onward to the global set of healthy addresses (' nginx.mynamespace.myfederation.svc.example.com.')
+
+The above set of DNS records is automatically kept in sync with the current state of health of all service shards globally by the Federated Service system.
+DNS resolver libraries (which are invoked by all clients) automatically traverse the hiearchy of 'CNAME' and 'A' records to return the correct set of healthy IP addresses. Clients can then select any one of the returned addresses to initiate a network connection (and fail over automatically to one of the other equivalent addresses if required).
+
+## Discovering a Federated Service
+
+### From Pods Inside your Federated Clusters
+
+By default, Kubernetes clusters come pre-configured with a cluster-local DNS server ('KubeDNS'), as well as an intelligently constructed DNS search path which together ensure that DNS queries like "myservice", "myservice.mynamespace", "bobsservice.othernamespace" etc issued by your software running inside Pods are automatically expanded and resolved correctly to the appropriate service IP of services running in the local cluster.
+
+With the introduction of Federated Services and Cross-Cluster Service Discovery, this concept is extended to cover Kubernetes services running in any other cluster across your Cluster Federation, globally.  To take advantage of this extended range, you use a slightly different DNS name (of the form "<servicename>.<namespace>.<federationname>", e.g. myservice.mynamespace.myfederation) to resolve Federated Services.  Using a different DNS name also avoids having your existing applications accidentally traversing cross-zone or cross-region networks and you incurring perhaps unwanted network charges or latency, without you explicitly opting in to this behavior.
+
+So, using our NGINX example service above, and the Federated Service DNS name form just described, let's consider an example: A Pod in a cluster in the `us-central1-f` availability zone needs to contact our NGINX service.  Rather than use the service's traditional cluster-local DNS name (```"nginx.mynamespace"```, which is automatically expanded to ```"nginx.mynamespace.svc.cluster.local"```) it can now use the service's Federated DNS name, which is ```"nginx.mynamespace.myfederation"```. This will be automatically expanded and resolved to the closest healthy shard of my NGINX service, wherever in the world that may be. If a healthy shard exists in the local cluster, that service's cluster-local (typically 10.x.y.z) IP address will be returned (by the cluster-local KubeDNS).  This is exactly equivalent to non-federated service resolution.
```

Comment:
We actually return a `CNAME` now and `KubeDNS` also appends the resolved `A` record to the response, but it is still worth documenting that behavior in this user guide.